### ISSUE TITLE
Creating/getting a contact returns empty contact object

### ISSUE THREAD

**Issue by ghostdevv** [mautic/mautic#13243]
### Mautic Version

5.0.x series

### PHP version

8.1.27

### What browsers are you seeing the problem on?

Not relevant

### What happened?

When I make a request to create/get a contact with the REST api it returns an empty object.

### How can we reproduce this issue?

![image](https://github.com/mautic/mautic/assets/47755378/526b6c76-5a61-425d-afa2-df8363e04281)

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by escopecz** [comment]
I'm not able to reproduce the issue as we discussed in https://mautic.slack.com/archives/C02L1C6UFHQ/p1705630439138129
```
curl -X "POST" "https://mautic-community.test/api/contacts/new" \
     -H 'Content-Type: application/json' \
     -u 'admin:Maut1cR0cks!' \     
     -d $'{
  "email": "john@doe.email"
}'
```

```json
{"contact":{"isPublished":true,"dateAdded":"2024-01-19T18:35:59+00:00","dateModified":null,"createdBy":14,"createdByUser":"Admin User","modifiedBy":null,"modifiedByUser":null,"id":15,"points":0,"color":null,"fields":{"core":{"title":{"id":66,"group":"core","label":"Title","alias":"title","type":"lookup","properties":{"list":["Mr","Mrs","Miss"]},"value":null,"normalizedValue":""},"firstname":{"id":67,"group":"core","label":"First Name","alias":"firstname","type":"text","properties":[],"value":null,"normalizedValue":""},"lastname":{"id":68,"group":"core","label":"Last Name","alias":"lastname","type":"text","properties":[],"value":null,"normalizedValue":""},"company":{"id":69,"group":"core","label":"Primary company","alias":"company","type":"text","properties":[],"value":null,"normalizedValue":""},"position":{"id":70,"group":"core","label":"Position","alias":"position","type":"text","properties":[],"value":null,"normalizedValue":""},"email":{"id":71,"group":"core","label":"Email","alias":"email","type":"email","properties":[],"value":"john@doe.email","normalizedValue":"john@doe.email"},"mobile":{"id":72,"group":"core","label":"Mobile","alias":"mobile","type":"tel","properties":[],"value":null,"normalizedValue":""},"phone":{"id":73,"group":"core","label":"Phone","alias":"phone","type":"tel","properties":[],"value":null,"normalizedValue":""},"points":{"id":74,"group":"core","label":"Points","alias":"points","type":"number","properties":[],"value":null,"normalizedValue":""},"address1":{"id":76,"group":"core","label":"Address Line 1","alias":"address1","type":"text","properties":[],"value":null,"normalizedValue":""},"address2":{"id":77,"group":"core","label":"Address Line 2","alias":"address2","type":"text","properties":[],"value":null,"normalizedValue":""},"city":{"id":78,"group":"core","label":"City","alias":"city","type":"text","properties":[],"value":null,"normalizedValue":""},"state":{"id":79,"group":"core","label":"State","alias":"state","type":"region","properties":[],"value":null,"normalizedValue":""},"zipcode":{"id":80,"group":"core","label":"Zip Code","alias":"zipcode","type":"text","properties":[],"value":null,"normalizedValue":""},"country":{"id":81,"group":"core","label":"Country","alias":"country","type":"country","properties":[],"value":null,"normalizedValue":""},"preferred_locale":{"id":82,"group":"core","label":"Preferred Locale","alias":"preferred_locale","type":"locale","properties":[],"value":null,"normalizedValue":""},"timezone":{"id":83,"group":"core","label":"Preferred Timezone","alias":"timezone","type":"timezone","properties":[],"value":null,"normalizedValue":""},"last_active":{"id":84,"group":"core","label":"Date Last Active","alias":"last_active","type":"datetime","properties":[],"value":null,"normalizedValue":""},"attribution_date":{"id":85,"group":"core","label":"Attribution Date","alias":"attribution_date","type":"datetime","properties":[],"value":null,"normalizedValue":""},"attribution":{"id":86,"group":"core","label":"Attribution","alias":"attribution","type":"number","properties":{"roundmode":4,"scale":2},"value":null,"normalizedValue":""},"website":{"id":87,"group":"core","label":"Website","alias":"website","type":"url","properties":[],"value":null,"normalizedValue":""}},"social":{"facebook":{"id":88,"group":"social","label":"Facebook","alias":"facebook","type":"text","properties":[],"value":null,"normalizedValue":""},"foursquare":{"id":89,"group":"social","label":"Foursquare","alias":"foursquare","type":"text","properties":[],"value":null,"normalizedValue":""},"instagram":{"id":90,"group":"social","label":"Instagram","alias":"instagram","type":"text","properties":[],"value":null,"normalizedValue":""},"linkedin":{"id":91,"group":"social","label":"LinkedIn","alias":"linkedin","type":"text","properties":[],"value":null,"normalizedValue":""},"skype":{"id":92,"group":"social","label":"Skype","alias":"skype","type":"text","properties":[],"value":null,"normalizedValue":""},"twitter":{"id":93,"group":"social","label":"Twitter","alias":"twitter","type":"text","properties":[],"value":null,"normalizedValue":""}},"personal":[],"professional":[],"all":{"id":"15","title":null,"firstname":null,"lastname":null,"company":null,"position":null,"email":"john@doe.email","mobile":null,"phone":null,"points":0,"address1":null,"address2":null,"city":null,"state":null,"zipcode":null,"country":null,"preferred_locale":null,"timezone":null,"last_active":null,"attribution_date":null,"attribution":null,"website":null,"facebook":null,"foursquare":null,"instagram":null,"linkedin":null,"skype":null,"twitter":null}},"lastActive":null,"owner":null,"ipAddresses":[],"tags":[],"utmtags":null,"stage":null,"dateIdentified":"2024-01-19T18:35:59+00:00","preferredProfileImage":null,"doNotContact":[],"frequencyRules":[]}}
```

But in this screenshot it's clear why it's not working. You have to use POST, not GET HTTP method.

**Comment by ghostdevv** [comment]
@escopecz In the screenshot I was using the get contact API :pray:

**Comment by mallezie** [comment]
I've encountered the same thing. And it seems to be related to the jms-bundle you are using. 

In mautic coming directly from the repository following packages are used. (See composer.lock file). 
jms/serializer-bundle 5.1.0
jms/serializer 3.23.0
jms/metadata 2.6.1

When installing through the composer-scaffold way, these packages are updated. 
jms/metadata  2.8.0
jms/serializer 3.29.1
jms/serializer-bundle 5.4.0

These break the contact api result. 

I'm still figuring out what is the exact culprit.

**Comment by mallezie** [comment]
Pinpointed the problem back to jms/serializer-bundle package. 
Only downgrading that one fixes the problem.

**Comment by mallezie** [comment]
And to be more specific jms/serializer-bundle 5.2.1 works, 5.3.0 breaks it. 
So it must be one of the changes from https://github.com/schmittjoh/JMSSerializerBundle/releases/tag/5.3.0

Where my suspicion is this is PR changing behavior for us. https://github.com/schmittjoh/JMSSerializerBundle/pull/920/
This PR enables new functionality from jms/serializer https://github.com/schmittjoh/serializer/pull/1469

One thing to note here, is this only occurs on mautic's installed through composer. When going directly from the main mautic repo (clone and set it up) this uses the bundled composer.lock file (where versions are fixed to good one). In a composer based environment this file is not used, and thus is open to updates of subpackages (like this one).

**Comment by mallezie** [comment]
Final note, this does not occur in M4, since that is constrained to jms/serializer bundle 4.x

**Comment by mollux** [comment]
@ghostdevv  

I dived into this, and there is indeed an issue, which I addressed in https://github.com/mautic/mautic/pull/13272
Can you confirm you're installing M5 via Composer and not via the zipped releases?

**Comment by mollux** [comment]
Unsure why this issues stays open, but it has ban resolved by #13272

Closing it manually
